### ISSUE TITLE
Ensure SOPs for reproducing the downloadable data are available and could work

### ISSUE THREAD

**Issue by zougloub** [spinalcordtoolbox/spinalcordtoolbox#1604]
Summary of downloadable data:

- precompiled binaries
- PAM50
- gm_model
- optic_models
- pmj_models
- deepseg_sc_models
- deepseg_gm_models

Of course some packages are built from confidential clinical data, but it doesn't mean the data can't be treated as opaque but following a known interface, and so processing scripts could be made available.

For each one, at least:
- Document the source dataset format
  eg. “MRI nifti slices in RPI orientation, downsampled to AxBxC voxels, with corresponding masks (2 for wm, 1 for gm, 0 for the rest), with file names `%d-{data,mask}.nii.gz`”
- Have a script that abstracts away the actual location of the data (parameter) and allows to regenerate computed data from datasets with the right format

Later:
- Perhaps create dummy datasets, so we'd have code coverage on that new code
- Have a script that allows to generate validation output, for coverage/unit "testing" of the corresponding method

**Comment by joshuacwnewton** [comment]
Some notes here:

> - gm_model

This no longer exists in SCT as of https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/3995.

> - precompiled binaries

This is now handled in a reproducible fashion using https://github.com/spinalcordtoolbox/spinalcordtoolbox-binaries.

> - deepseg_sc_models
> - deepseg_gm_models

These models are now ONNX models, and the creation of these models was documented here https://github.com/spinalcordtoolbox/deepseg_keras_onnx_conversion.

> - PAM50

The PAM50 template is at least in part being actively developed at https://github.com/spinalcordtoolbox/PAM50, and the `CHANGES.md` file contains some steps for regenerating the template files.

> - optic_models
> - pmj_models

However for these two, I'm not confident that we could retroactively go back and redocument missing procedures, because I think access to the source code is blocked by https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/3438.

That said, we're still very far from the SOP-style reproducibility that you describe. In fact... I'm not sure that kind of reproducibility is even possible to do retroactively. At the very least, the amount of work required would be... substantial, given the size of our team.

I think I would like to close this as a WONTFIX. Not because I want to, but because it's sort of the nature of academia + open source, especially for projects multiple years old at this point. I think we've done what we feasibly can do at this point. :sweat:
